### PR TITLE
LimboAPI and FastMOTD support for Velocity

### LINKED ISSUE #146: PacketEvents incompability
[Elytrium/FastPrepareAPI#3](https://github.com/Elytrium/FastPrepareAPI/issues/3)

### LINKED ISSUE #873: Can't access Velocity server due to LimboAPI and PacketEvents 2.4.0
I can't connect to 1.21 server because of LimboAPI library which is needed for LimboAuth+LimboFilter+LimboReconnect plugins. Please solve this...

### PR DESCRIPTION
**Pull request by UserNugget**
This pull request implements support for LimboAPI and FastMOTD.

Fixes https://github.com/Elytrium/LimboAPI/issues/146
Fixes https://github.com/retrooper/packetevents/issues/873

### CODE REVIEW

File: velocity/src/main/java/io/github/retrooper/packetevents/manager/InternalVelocityPacketListener.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/*
+ * This file is part of packetevents - https://github.com/retrooper/packetevents
+ * Copyright (C) 2022 retrooper and contributors
```
Comment (Bram1903) [inline velocity/src/main/java/io/github/retrooper/packetevents/manager/InternalVelocityPacketListener.java:3]: When either creating new files or updating old ones, the copyright year should be changed to the current year.

**Review by Bram1903 — CHANGES_REQUESTED** [review]

File: velocity/src/main/java/io/github/retrooper/packetevents/manager/InternalVelocityPacketListener.java (hunk 1 above)
Comment (UserNugget) [reply velocity/src/main/java/io/github/retrooper/packetevents/manager/InternalVelocityPacketListener.java]: Yep, changed it.

**Comment by retrooper** [comment]
Does this actually work?

**Comment by UserNugget** [comment]
> Does this actually work?

Yep, but require a bit cleanup.

**Comment by retrooper** [comment]
I'll happily add to the current progress, but getting some testers on this would be great.

**Comment by Bram1903** [comment]
I will give it a shot this evening.

**Comment by Bram1903** [comment]
Hello @UserNugget,

I attempted to test your branch with the latest Velocity build and the PacketEvents branch, but encountered an issue. Specifically, I'm receiving a `read timed out` error in my console and am unable to join the server. However, when I remove the LimboAPI plugin, everything functions as expected. 

Is there a dependency I might be missing (I couldn't find any listed) or does this need to be tested on a Linux-based operating system instead?

## Test Setup
- **Proxy**: Velocity 3.3.0-SNAPSHOT (git-9d25d309-b400) with PacketEvents (your branch, compiled) and LimboAPI
- **Backend**: Paper 1.21 (master@62ed302) with ViaVersion and ViaBackwards (5.0.1)
- **Client**: 1.20.6

## Console Screenshot
image

Kind regards,  
Bram

**Comment by UserNugget** [comment]
@Bram1903, you should use LimboAPI build from [LimboAPI PR](https://github.com/Elytrium/LimboAPI/actions/runs/9866763699) as it have some changes required for this PR to work.

**Comment by Bram1903** [comment]
> @Bram1903, you should use LimboAPI build from [LimboAPI PR](https://github.com/Elytrium/LimboAPI/actions/runs/9866763699) as it have some changes required for this PR to work.

Ahh, now it makes sense. Thanks for your quick reply. I just went offline for today, but will give this another look tomorrow morning!

**Comment by UserNugget** [comment]
I've tried to rethink how this PR might be used in the future, and looks like there are still some issues:
1) While FastPrepareAPI is used, only one PacketEvents instance will work properly
2) While FastPrepareAPI is used, all packets that was send directly via PacketEvents will be ignored

So looks like this PR still require some changes.

**Comment by bobhenl** [comment]
It would be good if they can be made soon so it's in the next release, so we don't need to use old build 😅

**Comment by rafi67000** [comment]
this PR can be closed, it's not needed anymore. LimboAPI works on it's own with
 ```yaml
      save-uncompressed-packets: true
      compatibility-mode: true
   ```

**Pull request closed without merging**